zml/tokenizer.zig

```diff
@@ -206,7 +272,7 @@ pub const Tokenizer = struct {
 
                             meta.assert(std.mem.eql(u8, cur_tok, input[input_off..][0..cur_tok.len]), "current token '{s}' not found in input string '{s}' !", .{ cur_tok, input[input_off..] });
                         }
-                        const concat_tokens = input[input_off..][0 .. cur_tok.len + next_tok.len];
+                        const concat_tokens = input[input_off..][0 .. cur_tok.len + if (tok_buff[i + 1] == self.special_tokens.unk) 1 else next_tok.len];
```

Comment:
this looks weird. What is it doing ?

Reply:
Previously if an `.unk` was being read near the end of the text being encoded, it would result in an index out of bounds as we would attempt to read `[0..cur_tok.len + next_tok.len]`, where `next_tok.len` is `"<unk>".len`, whereas the real character that was unknown only has a length of 1.

Reply:
ok. but then if next token is `<unk>` we shouldn't merge at all.